src/hatti/views/dataview.cljs

```diff
@@ -88,6 +88,39 @@
                              {:selected view})
       (put! shared/event-chan {:re-render view}))))
 
+(defn activate-settings-view! [view settings-section]
+  (let [view (keyword view)
+        settings-section (keyword settings-section)
+        views (-> @shared/app-state :views :all)
+        settings-views (-> @shared/app-state :views :settings :all)]
+    (when (contains? (set settings-views) settings-section)
+      (merge-into-app-state! shared/app-state
+                             [:views]
+                             {:selected view})
+      (merge-into-app-state! shared/app-state
+                             [:views :settings]
+                             {:active-tab settings-section})
+      (put! shared/event-chan {:re-render view}))))
+
+(defn activate-integrated-apps-view! [view settings-section app-type]
+  (let [view (keyword view)
+        settings-section (keyword settings-section)
+        settings-views (-> @shared/app-state :views :settings :all)]
+    (when (contains? (set settings-views) settings-section)
+      (transact!
+       shared/app-state
+       (fn [app-state]
+         (-> app-state
+             (assoc-in [:views] {:selected view})
```

Comment:
I'm pretty sure `assoc-in` has to work that way, tested it in the repl too, https://clojuredocs.org/clojure.core/assoc-in